internal/scorecard/alpha/scorecard.go

```diff
@@ -135,29 +177,36 @@ func ConfigDocLink() string {
 
 // waitForTestToComplete waits for a fixed amount of time while
 // checking for a test pod to complete
-func (o Scorecard) waitForTestToComplete(ctx context.Context, p *v1.Pod) (err error) {
-	waitTimeInSeconds := int(o.WaitTime.Seconds())
-	for elapsedSeconds := 0; elapsedSeconds < waitTimeInSeconds; elapsedSeconds++ {
+func (r PodTestRunner) waitForTestToComplete(ctx context.Context, p *v1.Pod) (err error) {
+
+	podCheck := func() (done bool, err error) {
 		var tmp *v1.Pod
-		tmp, err = o.Client.CoreV1().Pods(p.Namespace).Get(ctx, p.Name, metav1.GetOptions{})
+		tmp, err = r.Client.CoreV1().Pods(p.Namespace).Get(ctx, p.Name, metav1.GetOptions{})
 		if err != nil {
-			return fmt.Errorf("error getting pod %s %w", p.Name, err)
+			return true, fmt.Errorf("error getting pod %s %w", p.Name, err)
 		}
 		if tmp.Status.Phase == v1.PodSucceeded {
-			return nil
+			return true, nil
 		}
 
-		time.Sleep(1 * time.Second)
+		if ctx.Err() != nil {
+			// return a timeout error
+			return true, err
+		}
+		return false, nil
 	}
-	return fmt.Errorf("error - wait time of %d seconds has been exceeded", o.WaitTime)
+
+	err = wait.PollImmediateInfinite(time.Duration(1*time.Second), podCheck)
```

Comment:
We can pass the context to a different wait funciton to signal the polling to stop.
```suggestion
	err = wait.UntilWithContext(ctx, podCheck, time.Duration(1*time.Second))
```

Need to tweak the `podCheck` signature to take a context as well.

Reply:
changed this to wait.PollImmediateUntil() in latest commit.